src/pages/spawn/spawnHost/fields/UserTagsField.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect } from "react";
+import styled from "@emotion/styled";
+import { InstanceTag } from "gql/generated/types";
+import { convertArrayToObject } from "utils/array";
+import { UserTagRow } from "./userTagsField/UserTagRow";
+
+export interface UserTagsFieldStateType {
+  addedInstanceTags?: InstanceTag[];
+  deletedInstanceTags?: InstanceTag[];
+}
+
+interface UserTagsFieldProps {
+  data: UserTagsFieldStateType;
+  onChange: React.Dispatch<React.SetStateAction<any>>;
+  instanceTags: InstanceTag[];
```

Comment:
I think you need a handler to reset the state, so that the cancel button on the modal can use it right?

Reply:
It doesn't seem to reset the state completely. If you have an unsaved row, canceling then going back in leaves the row there